src/libraries/System.IO.Compression/tests/ZipArchive/zip_UpdateTests.Comments.cs

```diff
@@ -0,0 +1,189 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Xunit;
+
+namespace System.IO.Compression.Tests
+{
+    public partial class zip_UpdateTests : ZipFileTestBase
+    {
+        [Theory]
+        [MemberData(nameof(Encoding_GetData))]
+        public static void Update_ZipArchive_Comment(
+            Encoding constructorEncoding,
+            string originalComment, string expectedOriginalComment,
+            string updatedComment, string expectedUpdatedComment)
+        {
+            // The archive comment max size in bytes is determined by the internal
+            // ZipEndOfCentralDirectoryBlock.ZipFileCommentMaxLength const field.
+            // The current value of that field is set to ushort.MaxValue.
+
+            using var stream = new MemoryStream();
+            var testStream = new WrappedStream(stream, true, true, true, null);
+
+            // Create
+            using (var zip = new ZipArchive(testStream, ZipArchiveMode.Create, leaveOpen: true, constructorEncoding))
+            {
+                zip.Comment = originalComment;
+                Assert.Equal(expectedOriginalComment.Length, zip.Comment.Length);
+                Assert.Equal(expectedOriginalComment, zip.Comment);
```

Comment:
You don't need to match the length as a separate assert.
```suggestion
                Assert.Equal(expectedOriginalComment, zip.Comment);
```